app/models/task.rb

```diff
@@ -0,0 +1,27 @@
+class Task < ApplicationRecord
+  belongs_to :child
+  has_many :task_completions, dependent: :destroy
+
+  validates :name, presence: true
+  validates :frequency, inclusion: { in: %w[daily weekend specific_days] }
+
+  enum :time_of_day, { morning: 0, afternoon: 1, evening: 2 }
+
+  scope :ordered, -> { order(:time_of_day, :name) }
+
+  def due_on?(date)
+    case frequency
+    when "daily"
+      true
+    when "weekend"
+      [ 0, 6 ].include?(date.wday)  # Sunday = 0, Saturday = 6
```

Comment:
Lets make these constants, like, 

SUNDAY = 0
SATURDAY = 6
WEEKEND = [SATURDAY, SUNDAY]

and then we can do: 

WEEKEND.include?(date.wday)